### DISCUSSION TITLE
Debian 12 x64 start problem

### DISCUSSION THREAD

**Discussion by BreakHear7** [rejetto/hfs#610] [Q&A] [answered]
I try start hfs on Debian 12.5.0 x64, but I get error Illegal instruction and exit code 132.
after HINT: type "help" for help and befor using generic lib for crc32
run by /usr/bin/npx -y hfs@lates
npm@9.2.0
node v18.19.0
Python 3.12.1
uname -a:
Linux debian 6.1.0-21-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.90-1 (2024-05-03) x86_64 GNU/Linux

It worked good on x86 version of Debian

How i can fix it?

**Comment by rejetto** [comment]
try running with `--dev` at the end, and see if you get more info

**Reply by BreakHear7** [reply]
No, i don't get mor info

**Comment by rejetto** [comment]
also, see if the linux binary works for you, as it's supposed to be for x64

**Reply by BreakHear7** [reply]
I try https://github.com/rejetto/hfs/releases/download/v0.52.7/hfs-linux.zip, it's return Illegal instruction too

**Comment by rejetto** [comment]
i assume you are running without any plugin installed

**Reply by BreakHear7** [reply]
which plugin?i install node-gyp

**Reply by rejetto** [reply]
i was specifically thinking of thumbnails plugin, because it includes "native" stuff

**Reply by rejetto** [reply]
i meant HFS plugins

**Reply by BreakHear7** [reply]
it's first run, i just enter npx hfs@latest

**Reply by BreakHear7** [reply]
And it work on Debian x86 and not work on Debian x64

**Comment by rejetto** [comment]
the problem looks related to native/binary stuff.
the only 2 native things in hfs are the crc32 thing, that fails but it's expected (and seems to manage nicely from what you said), and another one that we can investigate, if you are willing to work a bit with me. :)

**Reply by BreakHear7** [reply]
OK, i'm ready)

**Comment by rejetto** [comment]
i decided to first try another way, easier for you.
I attempted a possible fix.
You can try it by running `npx hfs@exp`
let me know

**Reply by BreakHear7** [reply]
It's return same result - Illegal instruction

**Reply by rejetto** [reply]
that's very disappointing

**Reply by rejetto** [reply]
do you see message "fs-x-attributes not available" ?

**Reply by rejetto** [reply]
please paste your console content here

**Reply by BreakHear7** [reply]
07:45:59 HFS ~ HTTP File Server
07:45:59 © Massimo Melina <a@rejetto.com> - License https://www.gnu.org/licenses/gpl-3.0.txt
07:45:59 started 6/7/2024, 7:45:59 AM DEV
07:45:59 version 0.53.0-beta8.93
07:45:59 build 2024-06-06T15:36:43.711Z
07:45:59 cwd /root/.hfs
07:45:59 node v22.2.0
07:45:59 platform linux
07:45:59 pid 73780
07:46:00 config config.yaml
07:46:00 No config file, using defaults
access log file: logs/access.log
access error log: logs/access-error.log
07:46:01 HINT: type "help" for help
Illegal instruction

**Comment by rejetto** [comment]
your nodejs may be the problem.
try to installing a different version, like for different cpu.
even version 20 should be ok.

**Reply by BreakHear7** [reply]
i try it on 22.2.0 and on 18.19.0

**Reply by rejetto** [reply]
it's not the version number that worries me, but that you may be running a version that's not for your cpu

**Reply by BreakHear7** [reply]
also i try run on Debian x86 with node 18.19.0 and it's work there, but i need to use debian x64 now

**Reply by BreakHear7** [reply]
on Debian x86 i used node for x86, on Debian x64 i use node for x64

**Comment by rejetto** [comment]
see if running another software with npx works.
for example, you can try `npx serve`

**Reply by BreakHear7** [reply]
npx serve is running

**Reply by BreakHear7** [reply]
![Untitled](https://github.com/rejetto/hfs/assets/27731592/40aac9b9-46a9-4969-8360-c8fea5e96183)

**Comment by rejetto** [comment]
do you perhaps know some javascript?

**Reply by BreakHear7** [reply]
No

**Reply by BreakHear7** [reply]
But can try something

**Comment by BreakHear7** [comment]
Can you say exact versions of node and npm with wich it work?

**Reply by rejetto** [reply]
your version(s) is fine, i'm sure that's not the problem

**Comment by rejetto** [comment]
can you find where is npx storing HFS on your system?

**Reply by BreakHear7** [reply]
yes, /root/.npm/_npx

**Comment by rejetto** [comment]
ok. What i would do, is to put a few console lines in strategic places, that are involved in the starting of HFS, to help me understand exactly at what point it stops, to identify the problem,
but it's clumsy to do this here.
i'll see if i can provide more instructions

**Reply by BreakHear7** [reply]
what i can do?

**Reply by rejetto** [reply]
first reply here https://github.com/rejetto/hfs/discussions/610#discussioncomment-9699323

**Comment by gbrlfrc** [comment]
Hi @BreakHear7 ,
can you please post the output of  `cat /proc/cpuinfo` ?

**Reply by BreakHear7** [reply]
root@debian:~# cat /proc/cpuinfo
processor	: 0
vendor_id	: GenuineIntel
cpu family	: 15
model		: 4
model name	: Intel(R) Pentium(R) 4 CPU 3.00GHz
stepping	: 9
microcode	: 0x3
cpu MHz		: 3000.074
cache size	: 1024 KB
physical id	: 0
siblings	: 2
core id		: 0
cpu cores	: 1
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 5
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx lm constant_tsc pebs bts nopl cpuid pni dtes64 monitor ds_cpl cid cx16 xtpr lahf_lm pti
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs itlb_multihit mmio_unknown
bogomips	: 6000.14
clflush size	: 64
cache_alignment	: 128
address sizes	: 36 bits physical, 48 bits virtual
power management:

processor	: 1
vendor_id	: GenuineIntel
cpu family	: 15
model		: 4
model name	: Intel(R) Pentium(R) 4 CPU 3.00GHz
stepping	: 9
microcode	: 0x3
cpu MHz		: 3000.074
cache size	: 1024 KB
physical id	: 0
siblings	: 2
core id		: 0
cpu cores	: 1
apicid		: 1
initial apicid	: 1
fpu		: yes
fpu_exception	: yes
cpuid level	: 5
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx lm constant_tsc pebs bts nopl cpuid pni dtes64 monitor ds_cpl cid cx16 xtpr lahf_lm pti
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs itlb_multihit mmio_unknown
bogomips	: 6000.14
clflush size	: 64
cache_alignment	: 128
address sizes	: 36 bits physical, 48 bits virtual
power management:

**Reply by gbrlfrc** [reply]
I tested HFS on a same environment as yours, same node version.
```
gabriele@debian: ~$ uname -a
uname -a:
Linux debian 6.1.0-21-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.90-1 (2024-05-03) x86_64 GNU/Linux
```
and its working smoothly.
![image](https://github.com/rejetto/hfs/assets/61314427/0072870e-a09e-4e68-8e0f-4f611f24a9b1)

Im worried here the problem can be the hardware, the cpu you showed me is like 20 years old:

> Model of the CPU: Intel(R) Pentium(R) 4 CPU 3.00GHz
> Vendor: GenuineIntel
> Number of Cores: 1 core per physical processor, with Hyper-Threading enabled (making it appear as 2 logical processors)
> The date of release for the Intel Pentium 4 CPU 3.00GHz, specifically the one with the CPU family 15 and model 4, stepping 9, is likely around early to mid-2004. The Intel Pentium 4 CPUs with these specifications were introduced in this timeframe.

There can be many architectural differences, especially if you're using a version of Node.js that is relatively recent.
Unfortunately, even replicating an error in a similar context is very complex and I dont think is strictly related to HFS.

**Reply by BreakHear7** [reply]
but it work in Debian x86 on this cpu

**Reply by BreakHear7** [reply, marked as answer]
I start it successfully using schroot with Debian x86)

**Reply by rejetto** [reply]
wow, that's interesting
